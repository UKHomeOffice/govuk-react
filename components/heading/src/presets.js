import React from 'react';
import { LEVEL_SIZE } from '@govuk-react/constants';
import Heading from '.';

export const H1 = props => <Heading level={1} size={LEVEL_SIZE[1]} {...props} />;
export const H2 = props => <Heading level={2} size={LEVEL_SIZE[2]} {...props} />;
export const H3 = props => <Heading level={3} size={LEVEL_SIZE[3]} {...props} />;
export const H4 = props => <Heading level={4} size={LEVEL_SIZE[4]} {...props} />;
export const H5 = props => <Heading level={5} size={LEVEL_SIZE[5]} {...props} />;
export const H6 = props => <Heading level={6} size={LEVEL_SIZE[6]} {...props} />;
