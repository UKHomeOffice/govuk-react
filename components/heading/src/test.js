import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';

import Heading from '.';
import { H1, H2, H3, H4, H5, H6 } from './presets';

describe('Heading', () => {
  it('renders a Heading and all the H-level tags without crashing', () => {
    const example = 'example';
    const wrapper = <Heading>{example}</Heading>;
    const div = document.createElement('div');

    ReactDOM.render(wrapper, div);
    ReactDOM.render(<H1>{example}</H1>, div);
    ReactDOM.render(<H2>{example}</H2>, div);
    ReactDOM.render(<H3>{example}</H3>, div);
    ReactDOM.render(<H4>{example}</H4>, div);
    ReactDOM.render(<H5>{example}</H5>, div);
    ReactDOM.render(<H6>{example}</H6>, div);
  });

  it('allows custom string-based font size without crashing', () => {
    ReactDOM.render(<Heading size="SMALL">Test</Heading>, document.createElement('div'));
  });

  it('allows custom numeric GDS font size without crashing', () => {
    ReactDOM.render(<Heading size={16}>Test</Heading>, document.createElement('div'));
  });

  it('throws an error if an unsupported size is used', () => {
    const example = 'example';
    const div = document.createElement('div');

    expect(() => { ReactDOM.render(<Heading size={0}>{example}</Heading>, div); }).toThrow();
    expect(() => { ReactDOM.render(<Heading size={1}>{example}</Heading>, div); }).toThrow();
    expect(() => { ReactDOM.render(<Heading size={99999}>{example}</Heading>, div); }).toThrow();
    expect(() => { ReactDOM.render(<Heading size="test">{example}</Heading>, div); }).toThrow();
  });

  it('matches wrapper snapshot', () => {
    const example = 'example';
    const wrapper = <Heading>{example}</Heading>;

    expect(mount(wrapper)).toMatchSnapshot('wrapper mount');
  });
});
