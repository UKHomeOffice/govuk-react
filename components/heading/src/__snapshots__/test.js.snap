// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading matches wrapper snapshot: wrapper mount 1`] = `
.c0 {
  color: #0b0c0c;
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 700;
  font-size: 32px;
  line-height: 1.09375;
  display: block;
  margin-top: 0;
  margin-bottom: 30px;
}

@media print {
  .c0 {
    color: #000;
  }
}

@media print {
  .c0 {
    font-size: 32px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    font-size: 48px;
    line-height: 1.0416666666666667;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    margin-bottom: 50px;
  }
}

<Heading
  size="XLARGE"
>
  <src__StyledHeading
    size="XLARGE"
  >
    <h1
      className="c0"
      size="XLARGE"
    >
      example
    </h1>
  </src__StyledHeading>
</Heading>
`;
